CDCgov/prime-reportstream
# #1974 - add customerStatus to sender and receiver

This PR ...

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. *Include steps to test these changes*
This change simple adds a customerStatus value to each sender and receiver json blog in the database, and to the Sender and Receiver classes. Run src/scripts/db_one_off/1974_customer_status.sql against your local postgres database, then select a receiver and verify the jsonb object contains a customerStatus value.  Do the same for sender.

Build and run tests locally to ensure the classes are getting their data properly, and populating the customerStatus value.

## Changes
- Add customerStatus to Receiver.kt
- Add customerStatus to Sender.kt
- Update organizations.yml to provide initial values for customerStatus for all senders and receivers
- Create one-off script to populate existing sender and receiver values in the database

## Checklist

### Testing
- [x ] Tested locally?
- [ x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [? ] Downloaded a file from `http://localhost:7071/api/download`?
- [ -] Added tests?

### Process
- [N ] Are there licensing issues with any new dependencies introduced?
- [ Y] Includes a summary of what a code reviewer should test/verify?
- [ ?] Updated the release notes?
- [ N] Database changes are submitted as a separate PR?
- [ ?] DevOps team has been notified if PR requires ops support?

## Fixes
- #issue - List GitHub issues this PR fixes
- 1974

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue
- N/A

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - No
- Does this PR include changes in authentication and/or authorization of existing endpoints?
  - No
- Does this change introduce new dependencies that need vetting?
  - No 
- Does this change require changes to our infrastructure?
  - No
- Does logging contain sensitive data?
  - No
- Does this PR include or remove any sensitive information itself?
  - No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
  - Logging did not change as part of this PR
- Have you received any additional approvals needed for this change?
  - None required
